src/view/observer/mutationobserver.js

```diff
@@ -149,6 +149,11 @@ export default class MutationObserver extends Observer {
 			if ( mutation.type === 'childList' ) {
 				const element = domConverter.getCorrespondingViewElement( mutation.target );
 
+				// Do not collect mutations from UIElements.
+				if ( element && element.is( 'uiElement' ) ) {
```

Comment:
Or is it resolved by DOMConverter returning UI Element for any DOM element inside it?